### DISCUSSION TITLE
Is there any rule for mcrypt_get_iv_size deprecation?

### DISCUSSION THREAD

**Discussion by enkhtulga-gantsagaan** [rectorphp/rector#7463] [Q&A] [answered]
I am trying to upgrade php 5.4 to 87.2 by step by step (5.4 to 5.5, 5.5 to 5.6 etc).
But when i try to put mcrypt_get_iv_size deprecation function as example. rector doesn't make any change for this function. 
1. How to handle it? 
2.  And what is the name of deprecation function rule?

Here are example of restor.php:
```
return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/src'
    ]);

    // register a single rule
    // $rectorConfig->rule(InlineConstructorDefaultToPropertyRector::class);

    // define sets of rules
       $rectorConfig->sets([
           LevelSetList::UP_TO_PHP_72,
       ]);
    //    use current php version is difference from the one refactor to.
       $rectorConfig->phpVersion(PhpVersion::PHP_72);
};
```

**Comment by TomasVotruba** [comment, marked as answer]
Hi, thanks fyour question.

Could you share a PHP code, of what change exactly do you expect? 

The `mcrypt` was removed completelly and there is no replacement for it, neither migration path.
So it must be rather refactored to existing secure libraries manually.

**Reply by enkhtulga-gantsagaan** [reply]
Mm, Is there any rule for remove deprecated functions? What i want is to detect deprecated functions of php to upgrade php 5.4 to php 7.1.

**Reply by TomasVotruba** [reply]
Rector handles only A → B changes, not reporting. 

For detection or static analysis, better use https://github.com/PHPCompatibility/PHPCompatibility

**Reply by enkhtulga-gantsagaan** [reply]
Ok. thank you very much.
